### PR TITLE
add map modifier and the benefit when we research the technology other Civs have.

### PR DESCRIPTION
**Pull request by lishaoxia1985**

### CODE REVIEW

**Comment by yairm210** [comment]
Was this in original Civ?

**Comment by lishaoxia1985** [comment]
about this i only find Chinese web:
http://www.civclub.net/bbs/forum.php?mod=viewthread&tid=123976
about the number of map modifier i only find this( now i use that the Chinese web says):
https://www.carlsguides.com/strategy/civilization5/difficulty-settings.php

File: core/src/com/unciv/logic/civilization/TechManager.kt [hunk 1]
```diff
@@ -44,7 +45,20 @@ class TechManager {
         if (civInfo.isPlayerCivilization())
             techCost *= civInfo.getDifficulty().researchCostModifier
         techCost *= civInfo.gameInfo.gameParameters.gameSpeed.getModifier()
-        techCost *= 1 + (civInfo.cities.size -1 ) * 0.02f // each city increases tech cost by 2%, as per https://civilization.fandom.com/wiki/Science_(Civ5)
+        var techsResearchedKnownCivs = 0
+        civInfo.getKnownCivs().filter { it.isMajorCiv() }.forEach {
```
Comment (yairm210) [inline core/src/com/unciv/logic/civilization/TechManager.kt]: Use .count{} both here and in undefeatedCivs

File: core/src/com/unciv/logic/civilization/TechManager.kt [hunk 2]
```diff
@@ -44,7 +45,20 @@ class TechManager {
...
+            if (it.tech.isResearched(techName)) techsResearchedKnownCivs++
+        }
+        val undefeatedCivs= UnCivGame.Current.gameInfo.civilizations.filter { it.isMajorCiv() && !it.isDefeated() }.size
+        techCost /= 1 + techsResearchedKnownCivs / undefeatedCivs * 0.3f
```
Comment (yairm210) [inline core/src/com/unciv/logic/civilization/TechManager.kt]: Please add references in comments, for both this and the worldSizeModifier

Comment (lishaoxia1985) [reply core/src/com/unciv/logic/civilization/TechManager.kt]: ok, but i have not found a good english web to explain this, i am not sure whether adding Chinese web is a good idea or not.

Comment (yairm210) [reply core/src/com/unciv/logic/civilization/TechManager.kt]: Okay, better than nothing if we're going to add features based on that source

Comment (lishaoxia1985) [reply core/src/com/unciv/logic/civilization/TechManager.kt]: ok, but i have not found why it has no effort.

**Comment by lishaoxia1985** [comment]
now it's very good.
some suggestions: 
1. can you enlarge the map tile? now i think it is not fit. because city-bar is to big. the map tile is to small to place many food or other resource icons.
2. in techpickscreen, the progress bar in current researching techbutton can be deleted, it shows can be only in worldscreen. in the other hand, showing the scrollPane in techpickscreen is not a good idea too.

File: core/src/com/unciv/logic/civilization/TechManager.kt [hunk 3]
```diff
@@ -103,8 +114,7 @@ class TechManager {
     }
 
     fun nextTurn(scienceForNewTurn: Int) {
-        val currentTechnology = currentTechnologyName()
-        if (currentTechnology == null) return
+        val currentTechnology = currentTechnologyName() ?: return
```
Comment (yairm210) [inline core/src/com/unciv/logic/civilization/TechManager.kt]: Please don't use the Elvis operator, it's less understandable for people who aren't used to it

File: core/src/com/unciv/logic/civilization/TechManager.kt [hunk 4]
```diff
@@ -44,30 +46,39 @@ class TechManager {
...
+        }
+        techCost *= worldSizeModifier[0]
+        techCost *= 1 + (civInfo.cities.size -1) * worldSizeModifier[1]
         return techCost.toInt()
     }
 
     fun currentTechnology(): Technology? {
-        val currentTechnologyName = currentTechnologyName()
-        if (currentTechnologyName == null) return null
+        val currentTechnologyName = currentTechnologyName() ?: return null
```
Comment (yairm210) [inline core/src/com/unciv/logic/civilization/TechManager.kt]: And here

**Pull request merged**